File: datastructure/Tree/TreeDebugger.py
from typing import List
from TreeNode import TreeNode
class TreeDebugger:
    def build_levelorder_tree(self, level_order: List) -> TreeNode:
        n = len(level_order)
        root = TreeNode(level_order[0])
        nodes: List = []
        nodes.append(root)
        for i in range(1, n):
            new_node = None
            if level_order[i] is not None:
                new_node = TreeNode(level_order[i])
            parent_index = (i-1) // 2
            parent = nodes[parent_index]
            if i % 2 == 1:
                parent.left = new_node
            else:
                parent.right = new_node
            nodes.append(new_node)            
        return root
    
if __name__ =="__main__":
    level_order = [3, 9, 20, None, None, 15, 7]
    TreeDebugger().build_levelorder_tree(level_order)

#要匯入可以直接寫 from datastructure import TreeDebugger

File: datastructure/Tree/TreeDebugger.js
const TreeNode = require('./TreeNode');

class TreeDebugger {
    static buildLevelOrderTree(levelOrder) {
        if (levelOrder === null)
            return null
        const root = new TreeNode(levelOrder[0]);
        const nodes = []
        nodes.push(root)

        for (let i = 1; i < levelOrder.length; i++) {
            let newNode = null
            if (levelOrder[i] != null)
                newNode = new TreeNode(levelOrder[i])

            let parentIndex = Math.floor((i - 1) / 2)
            let parent = nodes[parentIndex]
            if (parent === null) {
                nodes.push(null)
                continue
            }
            if (i % 2 === 1) {
                parent.left = newNode;
            } else {
                parent.right = newNode;
            }
            nodes.push(newNode);

        }
        return root;
    }
    static main(){
        const levelOrder = [5, 4, 8, 11, null, 13, 4, 7, 2, null, null, null, 1]
        this.buildLevelOrderTree(levelOrder)
    }
}
if (require.main === module) {
    TreeDebugger.main();
}
module.exports = TreeDebugger;

/*
import
const TreeDebugger = require('../datastructure/Tree/TreeDebugger')
 */